src/firefly/js/tables/ui/BasicTableView.jsx

```diff
@@ -0,0 +1,233 @@
+                    showMaskNow();
+                    callbacks.onSort(sortInfoCls.toggle(cname).serialize());
+                }
+            };
+
+        const onFilter = ({fieldKey, valid, value}) => {
+                if (callbacks.onFilter) {
+                    if (valid && !filterInfoCls.isEqual(fieldKey, value)) {
+                        filterInfoCls.setFilter(fieldKey, value);
+                        showMaskNow();
+                        callbacks.onFilter(filterInfoCls.serialize());
+                    }
+                }
+            };
+
+        const colProps = pick(this.props, ['columns', 'data', 'selectable', 'selectInfoCls', 'callbacks', 'renderers']);
+        Object.assign(colProps, {columnWidths, filterInfoCls, sortInfoCls, showUnits, showFilters}, {onSort, onFilter, onSelect, onSelectAll});
+
+        const headerHeight = 22 + (showUnits && 12) + (showFilters && 20);
+        return (
+            <Resizable id='table-resizer' style={{position: 'relative', width: '100%', height: '100%', overflow: 'hidden'}} onResize={this.onResize}>
+                { textView ? <TextView { ...{columns, data, showUnits, heightPx, widthPx} }/> :
+                    <Table
+                        rowHeight={rowHeight}
+                        headerHeight={headerHeight}
+                        rowsCount={data.length}
+                        isColumnResizing={false}
+                        onColumnResizeEndCallback={this.onColumnResizeEndCallback}
+                        onRowClick={(e, index) => callbacks.onRowHighlight && callbacks.onRowHighlight(index)}
+                        rowClassNameGetter={this.rowClassName}
+                        scrollToRow={hlRowIdx}
+                        width={widthPx}
+                        height={heightPx}>
+                        { makeColumns(colProps) }
+                    </Table>
+                }
+                {showMask && <div style={{top: 0}} className='loading-mask'/>}
+                {isEmpty(data) && <div className='tablePanel_NoData'> No Data Found </div>}
+            </Resizable>
+        );
+    }
+}
+
+BasicTableView.propTypes = {
+    columns: PropTypes.arrayOf(PropTypes.object),
+    data: PropTypes.arrayOf(PropTypes.array),
+    hlRowIdx: PropTypes.number,
+    selectInfoCls: PropTypes.instanceOf(SelectInfo),
+    filterInfo: PropTypes.string,
+    sortInfo: PropTypes.string,
+    selectable: PropTypes.bool,
+    showUnits: PropTypes.bool,
+    showFilters: PropTypes.bool,
+    textView: PropTypes.bool,
+    rowHeight: PropTypes.number,
+    renderers: PropTypes.objectOf(
+        PropTypes.shape({
+            cellRenderer: PropTypes.func,
+            headRenderer: PropTypes.func
+        })
+    ),
+    callbacks: PropTypes.shape({
+        onRowHighlight: PropTypes.func,
+        onRowSelect: PropTypes.func,
+        onSelectAll: PropTypes.func,
+        onSort: PropTypes.func,
+        onFilter: PropTypes.func
+    })
+};
```

Comment:
Good example of `PropTypes.shape`. I have never used it before. I probably should start.